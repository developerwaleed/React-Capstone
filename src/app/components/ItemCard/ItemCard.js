import React from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import styles from './ItemCard.module.css';

const ProductComponent = () => {
  const products = useSelector((state) => state.allProducts.products);
  if (products.length === 0) {
    return (
      <div className={styles.loadingHome}>
        <div className={styles.Loader} />
      </div>
    );
  }
  const renderList = products.map((product) => {
    const { id, title, image, price, category } = product;
    return (
      <div className={styles.item} key={id}>
        <div className={styles.itemCard}>
          <div className={styles.itemImage}>
            <img src={image} alt={title} />
          </div>
          <div className={styles.itemDescription}>
            <h3 className={styles.itemName}>{title}</h3>
            <h4 className={styles.itemPrice}>${price}</h4>
            <h5 className={styles.itemCategory}>{category}</h5>
          </div>
        </div>
      </div>
    );
  });
  return <>{renderList}</>;
};

export default ProductComponent;
