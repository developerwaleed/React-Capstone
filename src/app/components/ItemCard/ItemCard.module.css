.loadingHome {
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 10rem 9.5rem;
}

.Loader {
  width: 50px;
  height: 50px;

  --c: radial-gradient(farthest-side, #fff 92%, #0000);

  background: var(--c) 50% 0, var(--c) 50% 100%, var(--c) 100% 50%,
    var(--c) 0 50%;
  background-size: 12px 12px;
  background-repeat: no-repeat;
  animation: s7 1s infinite;
}

.itemCard {
  min-height: 390px;
  max-width: 300px;
}

.itemImage img {
  width: 45%;
}

.itemName {
  font-size: 1.1rem;
}

.itemPrice {
  font-size: 1.7rem;
}

.itemCategory {
  font-size: 0.7rem;
}

.itemDescription {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    text-align: center;
    row-gap: 0.5rem;
    padding: 0.7rem 0.5rem;
  }
