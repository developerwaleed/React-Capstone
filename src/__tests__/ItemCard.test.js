import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import ItemCard from '../app/components/ItemCard/ItemCard';
import store from '../redux/configureStore';

it('renders properly', () => {
  const tree = renderer
    .create(
      <Router>
        <Provider store={store}>
          <ItemCard />
        </Provider>
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
