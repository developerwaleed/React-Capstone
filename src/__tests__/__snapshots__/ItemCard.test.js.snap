// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="loadingHome"
>
  <div
    className="Loader"
  />
</div>
`;
